operator/pkg/helmreconciler/rendering.go

```diff
@@ -41,6 +42,13 @@ import (
 	"istio.io/pkg/version"
 )
 
+var (
+	// objectCache holds the latest copy of each object applied by the controller, keyed by its Hash() function.
+	// Must be locked since ProcessManifest can be run concurrently.
+	objectCache   = make(map[string]*object.K8sObject)
```

Comment:
It seems like the cache should be associated with the ICP vs global.  Also, it doesn't look like objects are ever removed from the cache.

Reply:
even though there's no overlap between objects in different ICPs, the locking goes away, so this is a great idea - not sure why that didn't occur to me before as i was somehow fixated on cache correctness. 
regarding the object removal, is there a time we'd need/want to remove objects from the cache?

Reply:
As long as the cache is cleared at the start of any rendering operation, I don't see a need to clear it.  With the previous implementation, it seemed like the cache would never be cleaned, meaning nothing would be pruned (i.e. removed objects would still be in the cache).